src/url.js

```diff
@@ -134,17 +134,27 @@ exports.getUnrecognizedParametersInQueryString = function(queryString, options)
  *  - mapping : map short attributes to another value e.g. {q: 'query'}
  *  - moreAttributes : more values to be added in the query string. Those values
  *    won't be prefixed.
+ *  - safe : get safe urls for use in emails, chat apps or any application auto linking urls.
+ *  All parameters and values will be encoded in a way that it's safe to share them.
+ *  Default to false for legacy reasons ()
  * @return {string} the query string
  */
 exports.getQueryStringFromState = function(state, options) {
   var moreAttributes = options && options.moreAttributes;
   var prefixForParameters = options && options.prefix || '';
   var mapping = options && options.mapping || {};
+  var safe = options && options.safe || false;
   var invertedMapping = invert(mapping);
-  var partialStateWithEncodedValues = recursiveEncode(state);
+  var stateForUrl;
```

Comment:
I need var safe (for later usage and to follow in place conventions about options here), but agreed on the first part.